src/fair/metrics.py

```diff
@@ -299,3 +305,157 @@ def EF_X_agents(
                         envy_count += 1
                         break
     return envy_count
+
+
+# FUNCTIONS TO COMPUTE PAIRWISE MAXIMIN SHARE
+
+
+def create_sub_schedule(bundle_1: list[ScheduleItem], bundle_2: list[ScheduleItem]):
+    """Given two subsets of the set of items, create a new sub schedule considering the items in the union of both sets.
+    Capacities of the new schedule are determined by repetitions of the items in both bundles.
+    This function is necessary to compute PMMS metric. Creates a schedule with the items currently owned by two agents, in order to solve subproblem.
+
+    Args:
+        bundle_1 (list[ScheduleItem]): Items from class BaseItem
+        bundle_2 (list[ScheduleItem]): Items from class BaseItem
+
+    Returns:
+        new_schedule (list[ScheduleItem]): Items from class BaseItem, new reduced schedule
+        course_strings (list[str]): List of course strings of the new schedule
+        course (type[Course]): Course instance of the new schedule
+    """
+    sub_schedule = [*bundle_1, *bundle_2]
+    set_sub_schedule = sorted(list(set(sub_schedule)), key=lambda item: item.values[0])
+
+    course_strings = sorted([item.values[0] for item in set_sub_schedule])
+    course = Course(course_strings)
+    section = Section(sorted(list(set([item.values[3] for item in set_sub_schedule]))))
+    slot = sub_schedule[0].features[1]
+    weekday = sub_schedule[0].features[2]
+
+    features = [course, slot, weekday, section]
```

Comment:
The problem with using the original features, is that 'course' and 'section' consider the information of the entire schedule and not only the sub schedule. This means that when calling the constraint module, in particular:
`course_sect_constr = MutualExclusivityConstraint.from_items(new_schedule, course)`
the code breaks with the following error:
`return self.values[self.features.index(feature)]
ValueError: course: [160 ... 305] is not in list` 
I understand that to build the constraint, it is expecting values for the entire list of courses and not the reduced one. Let me know If there is something wrong in my reasoning, I didn't know if there was a better way of doing this.

Reply:
What are you trying to pull out of the valuation in that line that breaks? The access pattern I had in mind for valuations is in getting the value for a bundle, as is done [here](https://github.com/Fair-and-Explainable-Decision-Making/yankee-swap-allocation-framework/blob/main/tests/test_valuation.py#L28) for example.

Reply:
I don't think I understand what you are saying. The problem happens when creating the students for the sub problem, I need to pass them course_sect_constr and course_time_constr in order to create them. I originally tried passing the features as they are originally and got that error, so intuitively I thought it was probably because I was trying to build the constraints from the new_schedule and there is a mismatch with course. I didn't think it would be an issue to just alter course so that it only considers the courses in new_schedule, is this a bad solution? I don't understand what the proposed solution is.

Reply:
Oh sorry, I thought you were entering the command where the value error is raise, but that is just downstream. Can you check in the example with it breaking and I can take a look? Probably best to put it on a separate branch so that it doesn't prevent you from running experiments.

Reply:
Bug is gone, no need to redefine features anymore.